modules/core/shared/src/main/scala/weaver/Comparison.scala

```diff
@@ -0,0 +1,34 @@
+package weaver
+
+import cats.Eq
+import cats.Show
+import com.eed3si9n.expecty._
+
+trait Comparison[A] {
```

Comment:
This could potentially extend `Eq` as a `Comparison` instance does imply an `Eq` instance (and should satisfy `Eq` laws).

I'm not sure if the inheritance would play well with implicit searches.

Reply:
Interesting question. 

I think `Comparison` should not necessarily imply `Eq`, because you could have comparisons that verify if an instance `A` is contained in another `A` and I think that'd be a legitimate use. However there could be a more refined `EqComparison` that would imply `Eq`. In that case, the MTL treatment of having the implication reified as a `def eqInstance: Eq[A]` instead of OO-inheritance would probably be the right course of action, in order to avoid implicit search problems.

Reply:
That's an interesting case. In that vein, `Comparison` is a general predicate that has a string representation on failure? It doesn't have any laws associated with it.

Reply:
>  In that vein, Comparison is a general predicate that has a string representation on failure? It doesn't have any laws associated with it.

I think so, yes